File: 1598.cpp
#include<iostream>
#include<vector>
#include<string>

using namespace std;

class Solution {
public:
    int minOperations(vector<string>& logs) {
        int d = 0;
        for(string log : logs){
            d = log == "../" && d == 0 ? 0 : log == "../" ? --d : log != "./" ? ++d : d;
        }
        return d;
    }
};

int main(){
    Solution s;
    int n;
    cin >> n;
    vector<string> logs(n);
    for (int i = 0; i < n; i++) {
        cin >> logs[i];
    }
    cout << s.minOperations(logs) << endl;
    return 0;

}

File: python/1598.py
class Solution:
    def minOperations(self, logs: list[str]) -> int:
        d = 0
        for log in logs:
            d = 0 if log == '../' and d == 0 else d - 1 if log == '../' else d + 1 if log != './' else d
        return d
        

if __name__ == '__main__':
    s = Solution()
    logs = input().split()
    print(s.minOperations(logs))